appstore/developer_portal_api.py

```diff
@@ -0,0 +1,629 @@
+    allowed_fields = [
+        "developer_id"
+    ]
+
+    result = authed_request('GET', f"{config['REBBLE_AUTH_URL']}/api/v1/me")
+    if result.status_code != 200:
+        abort(401)
+    me = result.json()
+    if not me['is_wizard']:
+        return jsonify(error = "You are not a wizard", e = "permission.denied"), 403
+
+    try:
+        req = request.json
+    except Exception as e:
+        print(e)
+        return jsonify(error = "Invalid POST body. Expected JSON", e = "body.invalid"), 400
+
+    if req is None:
+        return jsonify(error = "Invalid POST body. Expected JSON", e = "body.invalid"), 400
+
+
+    for x in req:
+        if not x in allowed_fields:
+            return jsonify(error = f"Illegal field: {x}", e = "illegal.field"), 400
+
+    app = App.query.filter(App.id == appID).one_or_none()
+
+    if app is None:
+        return jsonify(error = "Unknown app", e = "app.notfound"), 404
+
+    changeOccured = False
+
+    if "developer_id" in req:
+            app.developer_id = req["developer_id"]
+            changeOccured = True
+            # return jsonify(error = "Failed to update developer ID. New ID does not exist", e = "newid.invalid"), 400
+
+
+    
+    if changeOccured:
+        try:
+            db.session.commit()
+            return jsonify(success = True, id = app.id, developer_id = app.developer_id)
+        except Exception as e:
+            print(e)
+            return jsonify(error = "Failed to update developer ID. Does new ID exist?", e = "body.invalid"), 400
+    else:
+        return jsonify(error = "Invalid POST body. Provide one or more fields to update", e = "body.invalid"), 400
+
+    
+@devportal_api.route('/wizard/app/<appID>', methods=['DELETE'])
+def wizard_delete_app(appID):
+    result = authed_request('GET', f"{config['REBBLE_AUTH_URL']}/api/v1/me")
+    if result.status_code != 200:
+        abort(401)
+    me = result.json()
+    if not me['is_wizard']:
+        return jsonify(error = "You are not a wizard", e = "permission.denied"), 403
+    
+    app = App.query.filter(App.id == appID).one_or_none()
+    if app is None:
+        return jsonify(error = "Unknown app", e = "app.notfound"), 404
+
+    App.query.filter(App.id == appID).delete()
+
+    db.session.commit()
+
+    if algolia_index:
+        algolia_index.delete_objects([algolia_app(app)])
+
+    return jsonify(success = True, id = app.id)
+
+@devportal_api.route('/wizard/app/<appID>', methods=['GET'])
+def wizard_get_s3_assets(appID):
+    result = authed_request('GET', f"{config['REBBLE_AUTH_URL']}/api/v1/me")
+    if result.status_code != 200:
+        abort(401)
+    me = result.json()
+    if not me['is_wizard']:
+        return jsonify(error = "You are not a wizard", e = "permission.denied"), 403
+    
+    app = App.query.filter(App.id == appID).one_or_none()
+    if app is None:
+        return jsonify(error = "Unknown app", e = "app.notfound"), 404
+
+    images = []
+    pbws = []
```

Comment:
Given these are uniqued at the end, consider making them sets instead of lists? Then they are inherently unique (though they do not have ordering).

Reply:
Sets aren't JSON serialisable natively in python